packages/vscode-extension/src/commonlib/azureLogin.ts

```diff
@@ -360,9 +379,21 @@ export class AzureAccountManager extends login implements AzureAccountProvider {
     const azureAccount: AzureAccount =
       vscode.extensions.getExtension<AzureAccount>("ms-vscode.azure-account")!.exports;
     AzureAccountManager.currentStatus = azureAccount.status;
+    if (AzureAccountManager.currentStatus === "LoggedIn") {
+      const subscriptioninfo = await this.readSubscription();
+      if (subscriptioninfo) {
+        this.setSubscription(subscriptioninfo.subscriptionId);
```

Comment:
What if the logged in account has no access to the selected subs?
Better to keep all the behaviors of current toolkit.
- When no user sign in but subs was selected
- When user sign in but no subs was selected
- When user sign in but selected subs mismatched

Reply:
When no user sign in but subs was selected
    No login means we can show nothing and do nothing.
When user sign in but no subs was selected
    We only show login information, if the user has only one sub we will select that and update subscription information file. If the user has more than 1 subs, we will show no sub selected.
When user sign in but selected subs mismatched
    For this scenario, solution will update env file when sign in or sign out. So i add signout clear file logic.

Reply:
`solution will update env file when sign in or sign out` interesting ... this means even the project is deployed, subs info could still be changed. That could be something to improve later.